src/app/content/search/hooks/index.ts

```diff
@@ -36,86 +38,89 @@ export const receiveSearchHook: ActionHookBody<typeof receiveSearchResults> = (s
   const state = services.getState();
   const {page, book} = selectContent.bookAndPage(state);
   const query = select.query(state);
-  const savedSearch = getSearchFromLocation(services.history.location);
+  const results = select.hits(state) || [];
 
   if (!page || !book) {
     return; // book changed while query was in the air
   }
 
-  const selectedResult = meta && meta.selectedResult ? meta.selectedResult : getFirstResult(book, payload);
+  const searchResultHit = meta && meta.searchScrollTarget && findSearchResultHit(results, meta.searchScrollTarget);
+  const selectedResult = searchResultHit && meta.searchScrollTarget
+    ? {result: searchResultHit, highlight: meta.searchScrollTarget.index}
+    : getFirstResult(book, payload);
 
-  if (!selectedResult) {
+  if (
+    // selectedResult may equal to null if api did not return any results
+    !selectedResult
+    // We are clearing selected result when requesting a new search so in the theory this should never happen
```

Comment:
this check was added previously so maybe there is a reasone for it

Reply:
i think the previous check was to make it not reload the search when clicking a figure link on a page when you have search results shown

Reply:
you seem to have removed the bit that looks at the target page, so we should make sure we didn't remove any important checks

Reply:
I've removed `page.id === stripIdVersion(targetPage.id)` check because page id is stored in selected result so this check should be enough
`isEqual(select.selectedResult(state), selectedResult)`